### PR TITLE
Enable test suite to run in parallel

### PR DESCRIPTION
**Pull request by pepeiborra**
To run the test suite in parallel with Cabal:

> cabal test --test-options="+RTS -N"

Locally, this runs in 58s in a Xeon with 56 logical cores

Importantly, this does not change CI
(unless stack passes +RTS -N secretly)

### CODE REVIEW

File: test/exe/Main.hs [hunk 1]
```diff
@@ -2107,14 +2106,14 @@ findDefinitionAndHoverTests = let
   tst (get, check) pos targetRange title = testSessionWithExtraFiles "hover" title $ \dir -> do
 
     -- Dirty the cache to check that definitions work even in the presence of iface files
-    liftIO $ runInDir dir $ do
```
Comment (wz1000) [inline test/exe/Main.hs:2110]: The point of this was to ensure that we have a different ghcide session, so that the shake graph is created anew using iface files.

File: test/exe/Main.hs [hunk 2]
```diff
@@ -2186,7 +2185,7 @@ findDefinitionAndHoverTests = let
   aaaL14 = Position 18 20  ;  aaa    = [mkR  11  0   11  3]
   dcL7   = Position 11 11  ;  tcDC   = [mkR   7 23    9 16]
   dcL12  = Position 16 11  ;
-  xtcL5  = Position  9 11  ;  xtc    = [ExpectExternFail,   ExpectHoverText ["Int", "Defined in ‘GHC.Types’"]]
+  xtcL5  = Position  9 11  ;  xtc    = [ExpectExternFail,   ExpectHoverText ["Int", "Defined in 'GHC.Types'"]]
```
Comment (wz1000) [inline test/exe/Main.hs:2188]: Why this change?

Comment (pepeiborra) [reply test/exe/Main.hs:2188]: `putStrLn` was crashing when trying to print the Unicode quotes (in test failures).

File: test/exe/Main.hs (hunk 1 above)
Comment (pepeiborra) [reply test/exe/Main.hs]: I thought it was a leftover. Reverted

**Review by wz1000 — APPROVED** [review]

**Comment by wz1000** [comment]
Nobody warned me about memory usage...
This ate through 12G of mem and 16G of swap like it was nothing on 12 logical core machine.
